src/components/AccountDropdown/AccountDropdown.js

```diff
@@ -68,9 +79,29 @@ class AccountDropdown extends Component {
     window.location.reload();
   }
 
+  onLogin = () => {
+    const keycloak = new Keycloak({
+      realm: "default",
+      clientId: "reaction-next-starterkit-client",
+      url: "http://localhost:8080/auth"
+    });
+
+    keycloak.init({ flow: "hybrid" }).then((authenticated) => authenticated).catch(() => {});
+
+    keycloak.login({ redirectUri: "http://localhost:4000/auth" })
+      .catch((error) => {
+        throw error;
+      });
```

Comment:
Even though this is MVP, after you get it stable, can you abstract this a bit?

- Add a file src/lib/auth/index.js
- Move the keycloak import and all of this onLogin code there. Do the init stuff immediately in that file.
- Export a generic `login` function from there that wraps `keycloak.login`
- Log an error for the init catch
- Get all of the URL/settings values from new ENV
- Import and call `login` function from `lib/auth` in this file

Reply:
Also, `redirectUri` should include something like `?r=/product/123`, where the value is the full current path + query string. And then AuthPage should use that for the redirect. In other words, you should end up back on the same page you're currently on. This will be important for checkout flow in particular.